backend/src/Repository/CaptainFinancialDuesEntityRepository.php

```diff
@@ -183,6 +185,26 @@ public function getLatestCaptainFinancialDuesByUserId(int $userId): CaptainFinan
 
             ->getQuery()
 
+            ->getOneOrNullResult();
+    }
+    //get the financial cycle to which the order belongs
+    public function getCaptainFinancialDuesByUserIDAndOrderId(int $userId, int $orderId): CaptainFinancialDuesEntity
```

Comment:
While the function may return one or null result, "?" have to be added to the return type.